### PR TITLE
Switch to 1-to-1 TestID to Result struct

### PR DESCRIPTION
**Pull request by sebrandon1**
Change the `[]claim.Result` to `claim.Result` when it comes to results being recorded in the claim file.

This involved changing all of the `testdata/` folders that contained sample JSON files.

### CODE REVIEW

**Comment by sebrandon1** [comment]
Marked WIP until I get the gradetool passing.

File: .github/workflows/pre-main.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ env:
   OCT_IMAGE_NAME: testnetworkfunction/oct
   OCT_IMAGE_TAG: latest
   GRADETOOL_IMAGE_NAME: testnetworkfunction/gradetool
-  GRADETOOL_IMAGE_TAG: latest
+  GRADETOOL_IMAGE_TAG: test1
```
Comment (sebrandon1) [inline .github/workflows/pre-main.yaml:17]: I can change this back once this merges: https://github.com/test-network-function/gradetool/pull/48

This one can only merge once the ginkgo_removal branch merges and we change the commit hardcoded in the go.mod.

**Review by edcdavid — COMMENTED** [review]
The claim format in the version.json should be updated to v0.2.0 to support this change

File: cmd/tnf/pkg/claim/claim.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ type TestCaseResult struct {
 }
 
 // Maps a test suite name to a list of TestCaseResult
-type TestSuiteResults map[string][]TestCaseResult
```
[the remark this answers is not in the document]
Comment (sebrandon1) [reply cmd/tnf/pkg/claim/claim.go:57]: Done.

**Review by edcdavid — APPROVED** [review]

**Pull request merged**